sklearn/linear_model/_ridge.py

```diff
@@ -194,13 +195,14 @@ def _solve_cholesky_kernel(K, y, alpha, sample_weight=None, copy=False):
             # Note: we must use overwrite_a=False in order to be able to
             #       use the fall-back solution below in case a LinAlgError
             #       is raised
-            dual_coef = linalg.solve(K, y, sym_pos=True, overwrite_a=False)
+            dual_coef = linalg.solve(K, y, sym_pos=True, overwrite_a=False,
```

Comment:
If `has_sw`, `y` can be the product of `y` and `sw`:

https://github.com/scikit-learn/scikit-learn/blob/cd6510f47ec82ad0c82eac4b8fba1ae81abf83c4/sklearn/linear_model/_ridge.py#L186

products are not always finite even if both `y` and `sw` are finite:

```python
import numpy as np

f_max = np.finfo(np.float64).max

# both finite arrays
X = np.full(2, f_max * .9)
sw = np.full(2, f_max * .9)

result = X * sw
np.isfinite(result).all()
# False
```